lib/SILOptimizer/Mandatory/DIMemoryUseCollectorOwnership.cpp

```diff
@@ -877,9 +877,30 @@ void ElementUseCollector::collectUses(SILValue Pointer, unsigned BaseEltNo) {
         addElementUses(BaseEltNo, PointeeType, User, DIUseKind::IndirectIn);
         continue;
 
+      
       // If this is an @inout parameter, it is like both a load and store.
-      case ParameterConvention::Indirect_Inout:
       case ParameterConvention::Indirect_InoutAliasable: {
+        // FIXME: The @inout_aliasable convention is used for indirect captures
+        // of both 'let' and 'var' variables. Using a more specific convention
+        // for 'let' properties like @in_guaranteed unfortunately exposes bugs
+        // elsewhere in the pipeline. A 'let' capture cannot really be mutated
+        // by the callee, and this is enforced by sema, so we can consider it
+        // a nonmutating use.
+        bool isLet = true;
+        
+        for (unsigned i = 0; i < TheMemory.NumElements; ++i) {
```

Comment:
Can you use none_of here instead? i.e.:

```
if (llvm::none_of(indices(TheMemory.NumElements),
                            [&](unsigned i) { return !TheMemory.isElementLetProperty(i) })) {
  addElementUses(...);
  continue;
}
```

Reply:
Thanks for the suggestion, but I don't find that easier to read.

Reply:
I am not going to push on this... but I just wanted to point out that I was just bringing up this section from the LLVM style guide:

https://www.llvm.org/docs/CodingStandards.html#turn-predicate-loops-into-predicate-functions

Reply:
Even that doesn't recommend using `none_of`, though, but splitting the loop out into a separate function,